homeassistant/components/here_travel_time/__init__.py

```diff
@@ -137,14 +120,57 @@ def _update(self) -> HERERoutingData | None:
                     ATTR_DURATION_IN_TRAFFIC: traffic_time / 60,
                     ATTR_DISTANCE: distance,
                     ATTR_ROUTE: response.route_short,
-                    ATTR_ORIGIN: ",".join(here_formatted_origin),
-                    ATTR_DESTINATION: ",".join(here_formatted_destination),
+                    ATTR_ORIGIN: ",".join(origin),
+                    ATTR_DESTINATION: ",".join(destination),
                     ATTR_ORIGIN_NAME: waypoint[0]["mappedRoadName"],
                     ATTR_DESTINATION_NAME: waypoint[1]["mappedRoadName"],
                 }
             )
+        except InvalidCoordinatesException as ex:
+            _LOGGER.error("Could not call HERE api: %s", ex)
         return None
 
+    def _prepare_parameters(
+        self,
+    ) -> tuple[list[str], list[str], str | None, str | None]:
+        """Prepare parameters for the HERE api."""
+        if self.config.origin_entity_id is not None:
+            origin = find_coordinates(self.hass, self.config.origin_entity_id)
+        else:
+            origin = self.config.origin
+
+        if self.config.destination_entity_id is not None:
+            destination = find_coordinates(self.hass, self.config.destination_entity_id)
+        else:
+            destination = self.config.destination
+        try:
+            here_formatted_destination = destination.split(  # type: ignore[union-attr]
```

Comment:
We'd rather handle the typing issue rather than ignore. Can we get to the root cause of this issue and resolve?

`find_coordinates` says it can return `str | None`. Is that why this fails?

Reply:
If that is right, then I would prefer check for none and raise the exception rather than catching AttributeError and ignoring the type, even though it duplicates the case where the schema is invalid.

Reply:
I replaced it with a `None` check and raise a exception as a control flow.